src/validation/parseFile.ts

```diff
@@ -225,14 +231,14 @@ const processSecurityIdentities = (
 
 const processMetadata = (
   caseInsensitiveDoc: CaseInsensitiveDocument<PrimitivesValues>,
-  documentBuilder: DocumentBuilder
+  documentBuilder: DocumentBuilder,
+  options?: ParseDocumentOptions
 ) => {
+  const metadata: Metadata = {};
   Object.entries(caseInsensitiveDoc.documentRecord).forEach(([k, v]) => {
-    documentBuilder.withMetadataValue(
-      k,
-      v! as Extract<PrimitivesValues, MetadataValue>
-    );
+    metadata[k] = v! as Extract<PrimitivesValues, MetadataValue>;
   });
```

Comment:
You don't use `k`, in that case, use `Object.values` instead.

Reply:
Not sure I get your point.
I need `k` to build my `Metadata` object right before calling the `withMetadata` method

Reply:
nvm, I misread.